### ISSUE TITLE
[Feature Request]Add support of model.eval() and model.train()

### ISSUE THREAD

**Issue by chenghuige** [tensorflow/tensorflow#22555]
For pytorch, you can use model.eval or mdoel.train to make sure different behavior like dropout.
For tf keras interface however if you write class using keras.Model, you still need to pass all training=training in call function, I think this is less elegant. 
So can we support model.train, model.eval ? when set model.train then all layer and model declared in __init__ is in training mode.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by wt-huang** [comment]
@chenghuige This is a valid point. We will note it down.

**Comment by Oktai15** [comment]
@wt-huang any update about this? Why did you close the issue?

**Comment by chenghuige** [comment]
@Oktai15  
from keras import backend as K
print K.learning_phase()
train_step.run(feed_dict={x: batch[0], labels: batch[1], K.learning_phase(): 1})

this might help

**Comment by milongo** [comment]
Are there any updates on this? It's not clear if the feature request was added or not.

**Comment by AbnerAI** [comment]
This is a reson why Pytorch surpass TF
